[bug] AuxPow functional test fails intermittently

# Bug Report

**Current behavior:**

Sometimes, qa/rpc-tests/auxpow.py fails with `not(-11 == 0)`

**Expected behavior**

The test doesn't fail randomly

**Dogecoin Core version**

1.14.0 - 1.14.4-dev

**Error log from CI**

```
auxpow.py:
Initializing test directory /tmp/tests9jm8az_/126
start_node: dogecoind started, waiting for RPC to come up
start_node: RPC successfully started
start_node: dogecoind started, waiting for RPC to come up
start_node: RPC successfully started
Mining blocks...
Stopping nodes

stderr:
Traceback (most recent call last):
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/auxpow.py", line 88, in <module>
    AuxPOWTest ().main ()
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/test_framework/test_framework.py", line 164, in main
    stop_nodes(self.nodes)
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/test_framework/util.py", line 385, in stop_nodes
    stop_node(node, i)
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/test_framework/util.py", line 380, in stop_node
    assert_equal(return_code, 0)
  File "/home/runner/work/dogecoin/dogecoin/qa/rpc-tests/test_framework/util.py", line 524, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(-11 == 0)

Pass: False, Duration: 2 s
```

--- a/src/validation.cpp
+++ b/src/validation.cpp
@@ -2504,9 +2504,6 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
     CBlockIndex *pindexMostWork = NULL;
     CBlockIndex *pindexNewTip = NULL;
     do {
-        boost::this_thread::interruption_point();
-        if (ShutdownRequested())
-            break;
 
         const CBlockIndex *pindexFork;
         ConnectTrace connectTrace;
@@ -2566,6 +2563,15 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
         if (pindexFork != pindexNewTip) {
             uiInterface.NotifyBlockTip(fInitialDownload, pindexNewTip);
         }
+
+        // Perform the shutdown detection to the end of the loop to prevent
+        // pindexNewTip being nil
+        boost::this_thread::interruption_point();
+        if (ShutdownRequested()) {
+            LogPrintf("ActivateBestChain: shutdown requested, breaking loop\n");
+            break;
+        }
+
     } while (pindexNewTip != pindexMostWork);
     CheckBlockIndex(chainparams.GetConsensus(pindexNewTip->nHeight));
 

